### ISSUE TITLE
Use VirtualThreads for Polaris TaskExecutor

### ISSUE THREAD

**Issue by jbonofre** [apache/polaris#411]

**Comment by snazy** [comment]
IMHO we should be really careful with Java virtual threads, because those are not a "just turn it on and everything's faster & better" thing. VTs come with quite some fine-print. In a nutshell: everything that runs on a VT **must** meet a bunch of criteria - otherwise the stability of the whole process is at risk, likely not in every environment ("number of CPUs").

Few posts that discuss this topic:
* [Quarkus - five things to know before using VTs](https://quarkus.io/blog/virtual-thread-1/#five-things-you-need-to-know-before-using-virtual-threads-for-everything)
* [Oracle - practical advice](https://docs.oracle.com/en/java/javase/20/core/virtual-threads.html#GUID-4A63E9EA-B16A-4633-87D8-C3DC08A3277C)
* [Netflix - locking issues w/ VTs](https://netflixtechblog.com/java-21-virtual-threads-dude-wheres-my-lock-3052540e231d)

**Comment by eric-maynard** [comment]
Agreed that we should look into optimizing this, though I don't know enough about virtual threads to have a strong opinion about the implementation. @snazy's concerns above seem valid.

Long-term, I am hoping that @flyrain's recent [proposal](https://docs.google.com/document/d/1Pd_mzZcfvnUvcH98IbwsIYf4eryet1lQDfclKYx-t-M/edit?tab=t.0#heading=h.7ic5c343eju1) will let us fully delegate this kind of work to an external service and make this problem moot.
